fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -590,46 +965,45 @@ ACTOR Future<Standalone<VectorRef<KeyValueRef>>> decodeRangeFileBlock(Reference<
 
 	Standalone<VectorRef<KeyValueRef>> results({}, buf.arena());
 	state StringRefReader reader(buf, restore_corrupted_data());
+	state Arena arena;
 
 	try {
-		// Read header, currently only decoding BACKUP_AGENT_SNAPSHOT_FILE_VERSION
-		if (reader.consume<int32_t>() != BACKUP_AGENT_SNAPSHOT_FILE_VERSION)
-			throw restore_unsupported_file_version();
-
-		// Read begin key, if this fails then block was invalid.
-		uint32_t kLen = reader.consumeNetworkUInt32();
-		const uint8_t* k = reader.consume(kLen);
-		results.push_back(results.arena(), KeyValueRef(KeyRef(k, kLen), ValueRef()));
-
-		// Read kv pairs and end key
-		while (1) {
-			// Read a key.
-			kLen = reader.consumeNetworkUInt32();
-			k = reader.consume(kLen);
-
-			// If eof reached or first value len byte is 0xFF then a valid block end was reached.
-			if (reader.eof() || *reader.rptr == 0xFF) {
-				results.push_back(results.arena(), KeyValueRef(KeyRef(k, kLen), ValueRef()));
-				break;
+		// Read header, currently only decoding BACKUP_AGENT_SNAPSHOT_FILE_VERSION or
+		// BACKUP_AGENT_ENCRYPTED_SNAPSHOT_FILE_VERSION
+		int32_t file_version = reader.consume<int32_t>();
+		if (file_version == BACKUP_AGENT_SNAPSHOT_FILE_VERSION) {
+			decodeKVPairs(&reader, &results);
+		} else if (file_version == BACKUP_AGENT_ENCRYPTED_SNAPSHOT_FILE_VERSION) {
+			// TODO (Nim): Remove once all work is finished, currently codepath should not trigger
+			ASSERT(false);
+			// Get cipher keys
+			SnapshotFileBackupCipherKeysCtx cipherKeyCtx = getCipherKeyCtx(arena);
+			// decode options struct
+			uint32_t optionsLen = reader.consumeNetworkUInt32();
+			const uint8_t* o = reader.consume(optionsLen);
+			StringRef optionsStringRef = StringRef(o, optionsLen);
+			EncryptedRangeFileWriter::Options options =
+			    ObjectReader::fromStringRef<EncryptedRangeFileWriter::Options>(optionsStringRef, IncludeVersion());
+
+			if (options.encryptionEnabled) {
```

Comment:
Shouldn't this be an assert and may be checked as the first thing in the routine?